src/java.base/share/classes/java/net/URI.java

```diff
@@ -2133,8 +2133,11 @@ private static String resolvePath(String base, String child,
         } else {
             StringBuilder sb = new StringBuilder(base.length() + cn);
             // 5.2 (6a)
-            if (i >= 0)
+            if (i >= 0) {
                 sb.append(base, 0, i + 1);
+            } else {
+                sb.append("/");
+            }
```

Comment:
This is wrong as it will cause  `URI.create("foo").resolve(URI.create("test"))` to return `"/test"` instead of `"test"`

Reply:
Your comment is correct. The behavior of specifying a relative URI as the base URI should not change, although rfc2396 recommended that the base URI be an absolute URI.  
I modified to add "/" only if the given base URI is an absolute URI.

Reply:
@dfuch 
Could you review this fix again, please？

Reply:
I will have a look but analyzing the proposed fix and thinking about the potential side effects will take me some time.

Reply:
@dfuch 
I merged tha latest source and modify this fix again to resolve conflict. Please check it again.

Reply:
@dfuch 
I'm sorry to bother you, is there anything wtong in this fix?